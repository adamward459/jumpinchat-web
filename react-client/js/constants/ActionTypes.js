/**
 * Created by Zaccary on 20/06/2015.
 */

export const SET_LAYOUT = 'SET_LAYOUT';
export const SET_SETTINGS_MODAL = 'SET_SETTINGS_MODAL';
export const SET_SETTINGS_ERROR = 'SET_SETTINGS_ERROR';


export const ROOM_LOAD = 'ROOM_LOAD';
export const ROOM_EMOJI = 'ROOM_EMOJI';
export const SET_ROOM_SETTINGS = 'SET_ROOM_SETTINGS';
export const SET_CLIENT_USER = 'SET_CLIENT_USER';
export const MESSAGE_ADD = 'MESSAGE_ADD';
export const MESSAGE_SEND = 'MESSAGE_SEND';
export const USER_CONNECT = 'USER_CONNECT';
export const USER_UPDATE_USERLIST = 'USER_UPDATE_USERLIST';
export const USER_DISCONNECT = 'USER_DISCONNECT';
export const USER_UPDATE_HANDLE = 'USER_UPDATE_HANDLE';
export const USER_UPDATE = 'USER_UPDATE';
export const USER_SET_NOTIFICATIONS = 'USER_SET_NOTIFICATIONS';
export const USER_SET_THEME = 'USER_SET_THEME';
export const USER_SET_UNREAD_MESSAGES = 'USER_SET_UNREAD_MESSAGES';
export const USER_SET_BROADCAST_QUALITY = 'USER_SET_BROADCAST_QUALITY';


export const SET_USER_LIST_OPTION_LIST = 'SET_USER_LIST_OPTION_LIST';
export const STREAM_REMOTE_DESTROY = 'STREAM_REMOTE_DESTROY';
export const STREAM_REMOTE_HANGUP = 'STREAM_REMOTE_HANGUP';
export const STREAM_REMOTE_RESUME = 'STREAM_REMOTE_RESUME';
export const STREAM_HANGUP_ALL = 'STREAM_HANGUP_ALL';
export const STREAM_RESUME_ALL = 'STREAM_RESUME_ALL';
export const STREAM_AUDIO_TOGGLE_MUTE = 'STREAM_AUDIO_TOGGLE_MUTE';
export const STREAM_AUDIO_TOGGLE_MUTE_ALL = 'STREAM_AUDIO_TOGGLE_MUTE_ALL';
export const STREAM_AUDIO_SET_GLOBAL_VOLUME = 'STREAM_AUDIO_SET_GLOBAL_VOLUME';
export const STREAM_LOCAL_CAN_BROADCAST = 'STREAM_LOCAL_CAN_BROADCAST';
export const STREAM_LOCAL_ADD = 'STREAM_LOCAL_ADD';
export const STREAM_LOCAL_DESTROY = 'STREAM_LOCAL_DESTROY';
export const STREAM_REMOTE_ADD = 'STREAM_REMOTE_ADD';
export const SET_STREAM_OPTIONS_STATE = 'SET_STREAM_OPTIONS_STATE';
export const SET_MESSAGE_SOUND = 'SET_MESSAGE_SOUND';
export const SET_SHOW_USERLIST = 'SET_SHOW_USERLIST';
export const SET_CHAT_INPUT_VALUE = 'SET_CHAT_INPUT_VALUE';
export const SET_WINDOW_IS_VISIBLE = 'SET_WINDOW_IS_VISIBLE';
export const SET_UNREAD_MESSAGES = 'SET_UNREAD_MESSAGES';
export const SET_CLIENT_PTT = 'SET_CLIENT_PTT';
export const SET_DEFAULT_PTT = 'SET_DEFAULT_PTT';
export const SET_REMOTE_STREAM_VOLUME = 'SET_REMOTE_STREAM_VOLUME';
export const STREAM_AUDIO_SET_ACTIVE = 'STREAM_AUDIO_SET_ACTIVE';
export const STREAM_SHOW_VOLUME_SLIDER = 'STREAM_SHOW_VOLUME_SLIDER';
export const STREAM_SET_LOADING = 'STREAM_SET_LOADING';
export const SET_LOCAL_AUDIO_ACTIVE = 'SET_LOCAL_AUDIO_ACTIVE';
export const RESUME_AUDIO_CONTEXT = 'RESUME_AUDIO_CONTEXT';
export const STREAM_SET_QUALITY = 'STREAM_SET_QUALITY';
export const SET_QUALITY_OPTIONS = 'SET_QUALITY_OPTIONS';
export const SET_MESSAGE_HISTORY = 'SET_MESSAGE_HISTORY';

// private messages
export const PM_START_CONVERSATION = 'PM_START_CONVERSATION';
export const PM_SELECT_CONVERSATION = 'PM_SELECT_CONVERSATION';
export const PM_REMOVE_CONVERSATION = 'PM_REMOVE_CONVERSATION';
export const PM_OPEN_MENU = 'PM_OPEN_MENU';
export const PM_CLOSE_CONVERSATION = 'PM_CLOSE_CONVERSATION';
export const PM_SET_CONVO_READ = 'PM_SET_CONVO_READ';

// Chat
export const CHAT_SET_USERS = 'CHAT_SET_USERS';
export const CHAT_SET_SETTINGS = 'CHAT_SET_SETTINGS';
export const CHAT_CLEAR_FEED = 'CHAT_CLEAR_FEED';
export const CHAT_SET_SCROLL = 'CHAT_SET_SCROLL';
export const CHAT_SET_SCROLL_FIXED = 'CHAT_SET_SCROLL_FIXED';

export const RESTORE_MESSAGE = 'RESTORE_MESSAGE';
export const SELECT_CHAT_TAB = 'SELECT_CHAT_TAB';
export const ADD_PRIVATE_MESSAGE = 'ADD_PRIVATE_MESSAGE';
export const SET_EMOJI_SEARCH = 'SET_EMOJI_SEARCH';
export const SET_EMOJI_RESULT_SELECTED = 'SET_EMOJI_RESULT_SELECTED';
export const SET_BANLIST = 'SET_BANLIST';

// modals
export const HANDLE_MODAL_STATE = 'HANDLE_MODAL_STATE';
export const BANLIST_MODAL_STATE = 'BANLIST_MODAL_STATE';
export const MEDIA_MODAL_STATE = 'MEDIA_MODAL_STATE';
export const MEDIA_MODAL_TYPE = 'MEDIA_MODAL_TYPE';
export const MEDIA_MODAL_SET_DEVICE_ID = 'MEDIA_MODAL_SET_DEVICE_ID';
export const MEDIA_MODAL_SET_AUDIO_PTT = 'MEDIA_MODAL_SET_AUDIO_PTT';
export const MEDIA_MODAL_LOADING = 'MEDIA_MODAL_LOADING';
export const MODAL_ERROR = 'MODAL_ERROR';
export const SET_JOIN_CONDITION_MODAL = 'SET_JOIN_CONDITION_MODAL';
export const SET_PROFILE_MODAL = 'SET_PROFILE_MODAL';
export const SET_IGNORE_MODAL = 'SET_IGNORE_MODAL';
export const SET_BAN_MODAL = 'SET_BAN_MODAL';


// sessions
export const SESSION_ID_SAVE = 'SESSION_ID_SAVE';
export const SESSION_IS_RECONNECTING = 'SESSION_IS_RECONNECTING';

export const ADD_NOTIFICATION = 'ADD_NOTIFICATION';
export const CLOSE_NOTIFICATION = 'CLOSE_NOTIFICATION';
export const PAUSE_NOTIFICATIONS = 'PAUSE_NOTIFICATIONS';
export const RESUME_NOTIFICATION = 'RESUME_NOTIFICATION';

export const CLIENT_UPDATE_HANDLE = 'CLIENT_UPDATE_HANDLE';

export const USER_SET = 'USER_SET';
export const USER_ACT_TOKEN_SET = 'USER_ACT_TOKEN_SET';
export const USER_LOG_IN = 'USER_LOG_IN';
export const HANDLE_MODAL_OPEN = 'HANDLE_MODAL_OPEN';
export const HANDLE_MODAL_CLOSE = 'HANDLE_MODAL_CLOSE';
export const HANDLE_MODAL_ERROR = 'HANDLE_MODAL_ERROR';
export const ACCOUNT_MODAL_STATE = 'ACCOUNT_MODAL_STATE';
export const INFO_MODAL_STATE = 'INFO_MODAL_STATE';
export const REPORT_MODAL_STATE = 'REPORT_MODAL_STATE';
export const SET_BROADCAST_RESTRICTED = 'SET_BROADCAST_RESTRICTED';
export const SET_EMOJI_PICKER = 'SET_EMOJI_PICKER';
export const INSERT_EMOJI = 'INSERT_EMOJI';
export const UPDATE_IGNORE = 'UPDATE_IGNORE';
export const USER_SET_PLAY_VIDEOS = 'USER_SET_PLAY_VIDEOS';

// Youtube
export const YOUTUBE_MODAL_STATE = 'YOUTUBE_MODAL_STATE';
export const YOUTUBE_SET_RESULTS = 'YOUTUBE_SET_RESULTS';
export const YOUTUBE_SET_VIDEO = 'YOUTUBE_SET_VIDEO';
export const YOUTUBE_SET_OPTIONS = 'YOUTUBE_SET_OPTIONS';
export const YOUTUBE_CLIENT_CLOSE_VIDEO = 'YOUTUBE_CLIENT_CLOSE_VIDEO';
export const YOUTUBE_ROOM_CLOSE_VIDEO = 'YOUTUBE_ROOM_CLOSE_VIDEO';
export const YOUTUBE_SET_VOLUME = 'YOUTUBE_SET_VOLUME';
export const YOUTUBE_SET_RESULTS_LOADING = 'YOUTUBE_SET_RESULTS_LOADING';
export const YOUTUBE_SET_VOLUME_CONTROL = 'YOUTUBE_SET_VOLUME_CONTROL';
export const YOUTUBE_SET_VIDEO_PAUSED = 'YOUTUBE_SET_VIDEO_PAUSED';
export const YOUTUBE_SET_VIDEO_RESUMED = 'YOUTUBE_SET_VIDEO_RESUMED';
export const YOUTUBE_SET_PLAYLIST = 'YOUTUBE_SET_PLAYLIST';
export const YOUTUBE_SET_PLAYLIST_ITEM_OPTIONS = 'YOUTUBE_SET_PLAYLIST_ITEM_OPTIONS';
export const YOUTUBE_CLEAR_SEARCH = 'YOUTUBE_CLEAR_SEARCH';
export const YOUTUBE_SET_VIDEO_SEEK = 'YOUTUBE_SET_VIDEO_SEEK';

// Push
export const PUSH_SET_MANAGER = 'PUSH_SET_MANAGER';
export const PUSH_SET_PUBLIC_KEY = 'PUSH_SET_PUBLIC_KEY';

export const PROFILE_SET = 'PROFILE_SET';
export const PROFILE_SET_NEW = 'PROFILE_SET_NEW';
export const PROFILE_LOADING = 'PROFILE_LOADING';
export const PROFILE_SET_IGNORE_ITEM = 'PROFILE_SET_IGNORE_ITEM';

// roles
export const ROLE_UPDATE = 'ROLE_UPDATE';
export const ROLES_SET = 'ROLES_SET';
export const ROLES_FETCH = 'ROLES_FETCH';
export const ROLES_SAVE = 'ROLES_SAVE';
export const ROLES_REMOVE = 'ROLES_REMOVE';
export const ENROLLMENTS_FETCH = 'ENROLLMENTS_FETCH';
export const ENROLLMENTS_SET = 'ENROLLMENTS_SET';
export const ENROLLMENT_USER_ADD = 'ENROLLMENT_USER_ADD';
export const ENROLLMENT_USER_ADD_FAILED = 'ENROLLMENT_USER_ADD_FAILED';
export const ENROLL_USER = 'ENROLL_USER';
export const UNENROLL_USER = 'UNENROLL_USER';
