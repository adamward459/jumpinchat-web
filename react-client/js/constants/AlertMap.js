export const ALERT_COLORS = {
  INFO: 'blue',
  SUCCESS: 'green',
  WARNING: 'yellow',
  ERROR: 'red',
};


export const ALERT_TYPES = {
  BANNER: 'banner',
  DIALOG: 'dialog',
  CHAT: 'chat',
};
