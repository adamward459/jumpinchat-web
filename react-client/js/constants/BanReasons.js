export default {
  BAN_BROADCAST_VIOLATION: 'Inappropriate broadcast',
  BAN_NUDITY: 'Nudity',
  BAN_RECORDING: 'Recording broadcasters',
  BAN_ABUSE: 'Abusing or promoting abuse',
  BAN_HARASSMENT: 'Harassing other members',
  BAN_UNDERAGE: 'Under age',
};
