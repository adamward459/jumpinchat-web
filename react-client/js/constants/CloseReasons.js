export const codes = {
  BAN_BROADCAST_VIOLATION: 'BAN_BROADCAST_VIOLATION',
  BAN_NUDITY: 'BAN_NUDITY',
  BAN_ABUSE: 'BAN_ABUSE',
  BAN_UNDERAGE: 'BAN_UNDERAGE',
};

export const reasons = {
  BAN_BROADCAST_VIOLATION: 'Inappropriate broadcasts',
  BAN_NUDITY: 'Nudity or encouraging nudity',
  BAN_ABUSE: 'Abusing or promoting abuse',
  BAN_UNDERAGE: 'Under age',
};
