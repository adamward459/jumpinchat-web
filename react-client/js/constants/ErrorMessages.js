/**
 * Created by Zaccary on 31/01/2016.
 */

export const ERR_NO_USER = 'no user found';
export const ERR_NO_ROOM = 'room does not exists';
export const ERR_NO_ROOM_ID = 'missing room ID';
export const ERR_NO_USER_LIST_ID = 'user not found in list';
export const ERR_NO_OP_TYPE = 'no moderator type';
export const ERR_NO_HANDLE = 'handle missing';
export const ERR_HANDLE_EXISTS = 'handle already in use';
export const ERR_HANDLE_VALIDATION = 'handle uses invalid characters';
export const ERR_HANDLE_LENGTH = 'handle too long';
export const ERR_USER_BANNED = 'banned';
export const ERR_NO_USER_SESSION = 'no user session';

export const ERR_YT_DEFAULT = 'Can\'t play video for some reason';
export const ERR_YT_BAD_PARAMS = 'Invalid parameters, can\'t play video';
export const ERR_YT_NO_HTML5 = 'Video can not be played in HTML5 player';
export const ERR_YT_NO_VIDEO = 'Video not found';
export const ERR_YT_NO_EMBED = 'Video can\'t be played while embedded';
