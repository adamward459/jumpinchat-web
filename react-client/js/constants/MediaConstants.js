export const defaultVideoConstraints = {
  width: 320,
  height: 240,
  frameRate: {
    ideal: 15,
    max: 30,
  },
};

export const goldVideoConstraints = {
  width: 960,
  height: 720,
  frameRate: {
    ideal: 30,
    max: 30,
  },
};

export const getVideoConstraints = config => ({
  width: {
    min: 320,
    ideal: config.dimensions.width,
    max: config.dimensions.width,
  },
  height: {
    min: 240,
    ideal: config.dimensions.height,
    max: config.dimensions.height,
  },
  frameRate: {
    ideal: config.frameRate,
    max: config.frameRate,
  },
});

export const goldVideoBitrate = 1024 * 1000;
