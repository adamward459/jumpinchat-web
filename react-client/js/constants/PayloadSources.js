export const SERVER_ACTION = 'SERVER_ACTION';
export const VIEW_ACTION = 'VIEW_ACTION';
