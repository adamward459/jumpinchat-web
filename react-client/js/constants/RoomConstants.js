/**
 * Created by Zaccary on 22/06/2015.
 */


export const RoomConstants = {
  maxMessages: 100,
};

export const chatTabs = {
  CHAT_FEED: 'CHAT_FEED',
  CHAT_PM: 'CHAT_PM',
  CHAT_USER_LIST: 'CHAT_USER_LIST',
};

export const layouts = {
  VERTICAL: 'VERTICAL',
  HORIZONTAL: 'HORIZONTAL',
};


export const PM_START_MESSAGE = 'Private conversation with';
export const BUSY_BROADCAST_THRESHOLD = 6;
