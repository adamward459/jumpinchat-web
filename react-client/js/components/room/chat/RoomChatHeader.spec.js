/* global it, beforeEach, describe */

import React from 'react';
import { shallow } from 'enzyme';
import { RoomChatHeader } from './RoomChatHeader.react';
import { chatTabs, layouts } from '../../../constants/RoomConstants';

describe('<RoomChatHeader />', () => {
  let roomChatHeader;
  beforeEach(() => {
    roomChatHeader = new RoomChatHeader();
  });

  describe('onMessageSoundChange', () => {
    beforeEach(() => {
      roomChatHeader.setRoomMessageSounds = jest.fn();
    });

    it('should set message sounds to false if enabled', () => {
      roomChatHeader.props = { messageSounds: true };
      roomChatHeader.onMessageSoundChange();
      expect(roomChatHeader.setRoomMessageSounds).toHaveBeenCalledWith(false);
    });

    it('should set message sounds to true if disabled', () => {
      roomChatHeader.props = { messageSounds: false };
      roomChatHeader.onMessageSoundChange();
      expect(roomChatHeader.setRoomMessageSounds).toHaveBeenCalledWith(true);
    });
  });

  describe('onToggleUserlist', () => {
    beforeEach(() => {
      roomChatHeader.setUserlist = jest.fn();
    });

    it('should set user list to false if enabled', () => {
      roomChatHeader.props = { showUserList: true };
      roomChatHeader.onToggleUserlist();
      expect(roomChatHeader.setUserlist).toHaveBeenCalledWith(false);
    });

    it('should set user list to true if disabled', () => {
      roomChatHeader.props = { showUserlist: false };
      roomChatHeader.onToggleUserlist();
      expect(roomChatHeader.setUserlist).toHaveBeenCalledWith(true);
    });
  });

  describe('selectFeedTab', () => {
    it('should call selectChatTab with `CHAT_FEED`', () => {
      roomChatHeader.selectChatTab = jest.fn();
      roomChatHeader.selectFeedTab();
      expect(roomChatHeader.selectChatTab).toHaveBeenCalledWith('CHAT_FEED');
    });
  });

  describe('selectPmTab', () => {
    it('should call selectChatTab with `CHAT_PM`', () => {
      roomChatHeader.selectChatTab = jest.fn();
      roomChatHeader.selectPrivateMessageTab();
      expect(roomChatHeader.selectChatTab).toHaveBeenCalledWith('CHAT_PM');
    });
  });

  describe('render', () => {
    let props;
    beforeEach(() => {
      props = {
        onToggleChat: jest.fn(),
        chatTab: chatTabs.CHAT_FEED,
        feedsMuted: false,
        messageSounds: true,
        camsDisabled: false,
        chatColors: ['foo'],
        room: {
          name: 'room',
          attrs: {
            owner: 'foo',
          },
          settings: {
            modOnlyPlayMedia: false,
          },
        },
        user: {
          user_id: 'foo',
          settings: {
            darkTheme: false,
          },
        },
        playYoutubeVideos: true,
        unreadConversations: 123,
        chatOpen: true,
        feedsHighDef: false,
        layout: layouts.VERTICAL,
        globalVolume: 0,
      };
    });

    it('should show unread conversation count in pm tab', () => {
      props.unreadConversations = 2;
      const wrapper = shallow(<RoomChatHeader {...props} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });
});
