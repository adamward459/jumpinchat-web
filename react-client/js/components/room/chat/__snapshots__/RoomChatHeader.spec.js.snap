// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomChatHeader /> render should show unread conversation count in pm tab 1`] = `
<div
  className="chat__Header"
>
  <div
    className="chat__HeaderMessageTabs"
  >
    <label
      className="button chat__HeaderOption chat__HeaderOption-toggleChat"
      title="Toggle chat feed"
    >
      <span
        aria-hidden="true"
        className="fa fa-chevron-down"
      />
      <input
        checked={true}
        className="jic-checkbox"
        onChange={[Function]}
        type="checkbox"
      />
    </label>
    <button
      className="button tabs__Item active"
      onClick={[Function]}
      type="button"
    >
      Chat
    </button>
    <button
      className="button tabs__Item"
      onClick={[Function]}
      type="button"
    >
      PMs
      <React.Fragment>
        Â 
        <span
          className="pill pill--blue pill--animated"
        >
          2
        </span>
      </React.Fragment>
    </button>
    <RoomChatShare
      roomName="room"
    />
  </div>
  <div
    className="chat__HeaderOptions"
  >
    <RoomCamOptions
      camsDisabled={false}
      chatColors={
        Array [
          "foo",
        ]
      }
      feedsCount={0}
      feedsHighDef={false}
      feedsMuted={false}
      globalVolume={0}
      layout="VERTICAL"
      modOnlyPlayMedia={false}
      playYoutubeVideos={true}
      roomHasOwner={true}
      roomName="room"
      settingsOptionsOpen={false}
      user={
        Object {
          "settings": Object {
            "darkTheme": false,
          },
          "user_id": "foo",
        }
      }
    />
    <button
      className="button chat__HeaderOption chat__HeaderOption--toggleUserlist button-blue"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-list-ul"
      />
    </button>
  </div>
</div>
`;
